test/cql-pytest/test_secondary_index.py

```diff
@@ -849,3 +849,157 @@ def test_index_quoted_names(cql, test_keyspace):
         for name in quoted_names:
             assert [(1,2)] == list(cql.execute(f'SELECT pk,ck FROM {table} WHERE {name} CONTAINS KEY 3'))
+
+# Tests basic operations on a static column index.
+def test_static_column_index(cql, test_keyspace):
+    schema = 'pk int, c int, s int STATIC, v int, PRIMARY KEY(pk, c)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f'CREATE INDEX ON {table}(s)')
+
+        # Insert
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (0, 0)')
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (1, 0)')
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (2, 1)')
+
+        assert [(0,),(1,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 0'))
+        assert [(2,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 1'))
+
+        # Update
+        cql.execute(f'UPDATE {table} SET s = 1 WHERE pk = 1')
+
+        assert [(0,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 0'))
+        assert [(1,),(2,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 1'))
+
+        # Partition delete
+        cql.execute(f'DELETE FROM {table} WHERE pk = 2')
+
+        assert [(0,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 0'))
+        assert [(1,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 1'))
+
+# Checks that clustering row deletions do not affect static columns.
+def test_static_column_index_unaffected_by_clustering_row_ops(cql, test_keyspace):
+    schema = 'pk int, c int, s int STATIC, v int, PRIMARY KEY(pk, c)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f'CREATE INDEX ON {table}(s)')
+
+        cql.execute(f'INSERT INTO {table} (pk, c, s, v) VALUES (0, 0, 42, 0)')
+        cql.execute(f'INSERT INTO {table} (pk, c, v) VALUES (0, 1, 10)')
+        cql.execute(f'INSERT INTO {table} (pk, c, v) VALUES (0, 2, 20)')
+        cql.execute(f'INSERT INTO {table} (pk, c, v) VALUES (0, 3, 30)')
+        cql.execute(f'INSERT INTO {table} (pk, c, v) VALUES (0, 4, 40)')
+
+        # We are not using SELECT DISTINCT because it is not implemented yet
+        # for queries that restrict a non-pk column. Therefore, `pk` appears
+        # multiple times in the result.
+
+        assert [(0,)]*5 == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 42'))
+
+        # Row delete
+        cql.execute(f'DELETE FROM {table} WHERE pk = 0 AND c = 4')
+        assert [(0,)]*4 == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 42'))
+
+        # Range delete
+        cql.execute(f'DELETE FROM {table} WHERE pk = 0 AND c >= 1 AND c < 3')
+        assert [(0,)]*2 == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 42'))
+
+        # Range delete, but this time get rid of all rows (static row should stay)
+        cql.execute(f'DELETE FROM {table} WHERE pk = 0 AND c >= 0 AND c <= 4')
+        assert [(0,)]*1 == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 42'))
+
```

Comment:
Something I would have liked to check here or in a separate test (maybe I'll see it below) is what happens if we have 4 rows in the same partition with s = 42, and now we set s = 3 - this should change move all four rows to match s=3 and nothing should match s=42 any more.

Reply:
I added a new test with the scenario you described: `test_static_column_index_all_clustering_rows_moved_by_static_column_update`